### DISCUSSION TITLE
Retronas SMB and Windows for Workgroups 3.11

### DISCUSSION THREAD

**Discussion by audiocrush** [retronas/discussions#69] [Q&A] [answered]
Hi there,

I've come across your project because I'm really desperate to figure out a way on how to proxy my retro software smb share on my TrueNAS to old protocol compatibility for my windows for workgroups 3.11 machines.
I did a setup where I used debian 11.3 and installed your retronas script.
Then I mounted my truenas retro software share via fstab
I installed samba with your script collection and then I duplicated your samba retronas.conf to my truenasproxy.conf
I made a new include in the smb.conf and changed the paths in truenasproxy.conf to the mountpoint of my truenas retro software share.

The main problem is though, be it your default config or me adding other parameters, trying to weaken security even further... WfW 3.11 finds the Server in the "map network drive" menu in the file explorer, but as soon as I doubleclick on it, it does not enumerate the shares.
So I enter it manually, click on ok and it asks me for a password...
So I enter no password: access denied
I enter the root password: access denied
I enter the password of the user I did set up retronas with: access denied.

No clue whats wrong here...
I mean it should at least enumerate the retronas default share and maybe also mine, before saying nope no way... :D

Any Ideas on this?
I read something about support for core protocol being dropped... but that shouldn't be a problem since WfW uses Lanman right? If I can draw aaaany conclusions from this super old article, then everything including NT1 one should be fine:
https://heyrick.eu/mynet/samba.html
It is so hard to get any info on this online.

Cheers audiocrush

**Comment by sairuk** [comment]
If I am understanding you correctly you do not want to use RetroNAS, instead want to apply the RetroNAS samba configuration to TrueNAS to try and achieve WfW compatiblity. 

I'm sure you'll understand TrueNAS has an entire community for support already. They would be better served to answer why TrueNAS wont work where RetroNAS does. We are not a TrueNAS support forum.

https://www.truenas.com/community/

**Reply by audiocrush** [reply]
No that is not correct.
I installed RetroNAS on a Debian machine in its default config.
Installed and enabled the samba service in the retronas menu and tried to connect to it using WfW 3.11, which didn't work out of the box and also not with modifications.

The truenas just serves as the storage where the data is ultimately stored.

By the way, using RetroNAS for achieving compatibility to other things is intrinsically the only reason these people even make RetroNAS in the first place in case you didn't realize it yet.
Why else would you invest all that energy in creating it from the ground up if you could just use everything else without problems?

**Reply by sairuk** [reply]
> No that is not correct. I installed RetroNAS on a Debian machine in its default config. Installed and enabled the samba service in the retronas menu and tried to connect to it using WfW 3.11, which didn't work out of the box and also not with modifications.

OK looks like I misunderstood your use case, thanks for clarifying

**Reply by audiocrush** [reply]
No worries.
I've looked at your ansible script install_samba.yml
It seems to set up a few things that, at least to my mind, should provide the compatibility to WfW 3.11 but I just don't know if it is enough or not. The thing is, which I'm really not sure about, It is very unclear to me if WfW 3.11 ALWAYS wants to provide a password when accessing a share and if so, it does it in plaintext auth or that weird windows password encryption auth (which is incompatible with /etc/passwd format in unix by the way.
I have 2 WfW 3.11 machines that we could use for testing purposes and maybe we can identify the cause of it not working together.

Later in the weekend I'm also going to install Microsoft Lan Manager with an IP stack on another 386 of mine to see if that maybe works and provide you with the outcome.

**Reply by audiocrush** [reply]
My current setup on the WFW side is:
I have the same workgroup name as the samba server (workgroup)
I'm logging on with the same username on the WFW machine as I'm also using for the samba server on RetroNAS and I also used for setting up RetroNAS, and I also set up the same password in WfW as on the RetroNAS machine.

I also found that using these parameters in the share config will increase compatibility with WfW 3.11 (in case you get it to connect in the first place :D):
case sensitive = no
default case = upper
preserve case = no
short preserve case = no
mangle case = yes
mangled names= yes

**Comment by audiocrush** [comment, marked as answer]
OK 
It seems I got it to work with a combo of things:
Not sure if Step 1 is relevant but:
First use WfW 3.11 Disk 8 and expand admincfg.exe onto your computer:
expand A:\admincfg.ex_ C:\windows\admincfg.exe
Then go into windows, and start admincfg.exe using file explorer
There you can click onpasswords and disable password cacheing (this is apparently needed when accessing shares with user level security and samba has dropped share level security support already if I got that correctly)

Next this is the samba config that got it right for me:
smb.conf:
[global]
bind interfaces only = yes
# enp0s5 = RetroNET VLAN12
interfaces = enp0s5
min protocol = CORE
netbios name = retronas
lanman auth = yes
client lanman auth = yes
ntlm auth = yes
map to guest = bad user
workgroup = WORKGROUP
client ntlm auth = yes
ntlmv2 auth = yes
client ntlmv2 auth = yes
unix extensions = no
allow insecure wide links = yes
[retronas]
include = /etc/samba/retronas.conf
[software]
include = /etc/samba/TruenasProxy.conf

TruenasProxy.conf
comment = Software
path = /mnt/FreenasProxy
guest ok = yes
public = yes
browseable = yes
writeable = yes
force user = audiocrush
force group = audiocrush
create mask = 0775
directory mask = 0775
follow symlinks = yes
wide links = yes
socket options = TCP_NODELAY IPTOS_LOWDELAY
case sensitive = no
default case = upper
preserve case = no
short preserve case = no
mangle case = yes
mangled names= yes

absolutely flawless :)
Just the 8.3 filename stuff doesn't seem to work properly yet.
but I'll keep you updated in case I find something out.

**Reply by sairuk** [reply]
Great thanks for the update
